warrior/Framework/Utils/data_Utils.py

```diff
@@ -901,10 +899,15 @@ def validate():
     keys = key.split('.')
     value = get_object_from_datarepository(keys[0])
     for k in keys[1:]:
+        if k not in value:
+            print_warning("key {} not present in data repository\n".format(key))
         value = value[k]
```

Comment:
Hi @sugan19,

In line 902, you have added an ```if``` statement to check if k does not exist in value. If it does not, then you are printing a warning - which works.

Now, once we exit that ```if``` block, we reach line 904, where we are assigning ```value``` the value of ```value[k]```; but if ```k``` does not exist in ```value```, line 904 will throw an error. 

I would suggest that the statement ```value = value[k]``` be put inside an ```else``` block so that this situation is avoided.

Thanks.

Reply:
@sanikakulkarni  Addressed you comments.